fontforgeexe/fontview.c

```diff
@@ -656,16 +656,19 @@ int _FVMenuSaveAs(FontView *fv) {
     }
 #endif
 
-    ret = GWidgetSaveAsFileWithGadget8(_("Save as..."),temp,0,NULL,
+    utemp = utf82u_copy(temp);
+    ret = GWidgetSaveAsFileWithGadget((unichar_t*)_("Save as..."),utemp,0,NULL,
 				       _FVSaveAsFilterFunc, FilenameFunc,
 				       &gcd );
-    free(temp);
+    free(temp); free(utemp);
     if ( ret==NULL )
 return( 0 );
-    filename = utf82def_copy(ret);
-    free(ret);
+    filename = u2utf8_copy(ret);
 
-    if(!(endswithi( filename, ".sfdir") || endswithi( filename, ".sfd")))
+    const unichar_t *u_sfdir = utf82u_copy(".sfdir");
+    const unichar_t *u_sfd = utf82u_copy(".sfd");
+
+    if(!(u_endswith( ret, u_sfdir ) || u_endswith( ret, u_sfd )))
```

Comment:
Does anyone know why I have to do this? Why can't I just:

```c
if(!(u_endswith( ret, (unichar_t*) ".sfdir" ) || u_endswith( ret, (unichar_t*) ".sfd" )))
```

Reply:
C is a low-level language and you're casting a pointer here. A cast can modify the format of the variable itself but in the case of a pointer, not what is pointed to. If what is pointed to needs modification to match the new pointer type the cast should fail.

Reply:
@skef OK but why does `(unichar_t*) "example"` work in other areas of the codebase, e.g. metricsview.c:816?

Reply:
Do you have a link to that @ctrlcctrlv ? I saw this: https://github.com/fontforge/fontforge/blob/master/fontforgeexe/metricsview.c#L816

In that case most likely that's just some specialty in gdraw where inside gdraw it knows to treat it not as a unichar_t but as a char or something like that.

Reply:
If you mean `(unichar_t *) _("Name:")` it's because `_()` is actually a function that takes a `char *` as its parameter, used for internationalization. It (apparently) already returns a wide string.

Reply:
Hmm...OK, thanks guys, just seems odd to me it's so hard to declare a constant (unichar_t*)